crates/matrix-sdk-sqlite/src/state_store.rs

```diff
@@ -446,7 +446,16 @@ impl SqliteStateStore {
     }
 
     async fn acquire(&self) -> Result<SqliteAsyncConn> {
-        Ok(self.pool.get().await?)
+        let conn = self.pool.get().await?;
+
+        // Specify a busy timeout so that operations are automatically retried, in case
+        // the database was marked as locked, which can happen under very
+        // peculiar circumstances in WAL mode.
+        //
+        // See also https://www.sqlite.org/wal.html#sometimes_queries_return_sqlite_busy_in_wal_mode.
+        conn.execute_batch("PRAGMA busy_timeout = 2000;").await?;
```

Comment:
Would be nice to specify that this is 2000 milliseconds.

It's not clear without searching through the docs.

I would probably define a `const` `Duration` for this, so the type tells us which unit we're using.

Reply:
Do we want to add this `PRAGMA` to other databases?